src/middlewares/stats.js

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable require-jsdoc */
+
+import jwt from 'jsonwebtoken';
+
+import StatsService from '../services/db.service';
+import { notFound, error } from '../helpers/responses';
+
+const { getStat } = StatsService;
+
+class statsWare {
+  static async checkStats(req, res, next) {
+    if (!req.auth) {
+      return error('you are not logged in').send(res);
+    }
+    const readerId = req.auth.id;
+    const stats = await getStat({ readerId }, 'Stats');
+    if (!stats.length) {
+      return notFound('reading stats').send(res);
+    }
+    next();
+  }
+
+  static async saveStat(req, res, next) {
+    try {
+      let token = req.headers['x-access-token'] || req.headers.authorization;
+      token = token.slice(7, token.length);
+      jwt.verify(token, process.env.SECRET_KEY, (err, decode) => {
+        req.auth = decode;
+        next();
+      });
+    } catch (err) {
+      next();
+    }
+  }
+}
+
+export default statsWare;
```

Comment:
What does the statsWare middleware do @Anguandia

Reply:
Thanks for the question Patrick. 
  - On the get article route, the saveStat method of statsWare does the scanning for the auth header in the request and if found, decodes the token and appends an auth property to the request so that the getOne method in the article controller can extract the current userId from this auth property for constituting this requests statistic.
  - The same function in the users/getstats route identifies the user whose statistics should be fetched; pretty much like the auth function but will more streamlined to tis particular function, only reminding the requester to login if not done
  - The checkStat method of the statsWare middleware will check the stats table for user stats after the saveStat confirms authenticated user. If not found, will return the 404 response before the request hits the stats controller. In this way, the stats controller is only left with processing and returning responses for confirmed viable stats get requests